cmd/util/nodehelpers/logging.go

```diff
@@ -0,0 +1,104 @@
+package nodehelpers
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/cmd/genericconf"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+var globalFileHandlerFactory = fileHandlerFactory{}
+
+type fileHandlerFactory struct {
+	writer  *lumberjack.Logger
+	records chan *log.Record
+	cancel  context.CancelFunc
+}
+
+// newHandler is not threadsafe
+func (l *fileHandlerFactory) newHandler(logFormat log.Format, config *genericconf.FileLoggingConfig, pathResolver func(string) string) log.Handler {
+	l.close()
+	l.writer = &lumberjack.Logger{
+		Filename:   pathResolver(config.File),
```

Comment:
I don't like moving function pointers around where is isn't a great reason, it's much harder to read code that calls to a function passed as parameter. newHandler can certainly just take the final path instead.
Not entirely certain about initLog. Possiby - instead of passing a function just pass a "folder to use as base if config.File is not relative". But again - not certain.

Reply:
I changed `newHandler` arguments to pass a base directory path instead of a function pointer.
However I left the path resolving function pointer in `genericconfig.InitLog` to be able to easily use `go-ethereum/node.Config.ResolvePath` for resolving logs location in `cmd/nitro`.